src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEquals.cs

```diff
@@ -20,11 +16,12 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     /// </summary>
     internal sealed class SynthesizedRecordEquals : SynthesizedRecordOrdinaryMethod
     {
-        private readonly PropertySymbol _equalityContract;
+        private readonly PropertySymbol? _equalityContract;
 
-        public SynthesizedRecordEquals(SourceMemberContainerTypeSymbol containingType, PropertySymbol equalityContract, int memberOffset, BindingDiagnosticBag diagnostics)
+        public SynthesizedRecordEquals(SourceMemberContainerTypeSymbol containingType, PropertySymbol? equalityContract, int memberOffset, BindingDiagnosticBag diagnostics)
             : base(containingType, WellKnownMemberNames.ObjectEquals, hasBody: true, memberOffset, diagnostics)
         {
+            Debug.Assert(equalityContract is null == containingType.IsStructType());
```

Comment:
>containingType.IsStructType() [](start = 53, length = 29)

I think we should be consistent in the way we check for record class vs. record struct. It feels like relying on a special property would be preferred, that way we will be able to quikly locate all the placess that perform the check with the purpose to distinguish kinds of records.

Reply:
I'm planning to merge `IsRecord` and `IsRecordStruct`. So I'll align on TypeKind check for class vs. struct.

---
In reply to: [597069643](https://github.com/dotnet/roslyn/pull/51900#discussion_r597069643) [](ancestors = 597069643)

Reply:
We could keep an `IsRecordStruct => IsRecord && TypeKind == Struct`, but then the `IsRecord` check is redundant. Is that what you were thinking?

---
In reply to: [597866787](https://github.com/dotnet/roslyn/pull/51900#discussion_r597866787) [](ancestors = 597866787,597069643)

Reply:
> We could keep an IsRecordStruct => IsRecord && TypeKind == Struct, but then the IsRecord check is redundant. Is that what you were thinking?

I would leave implementation of IsRecordStruct as is, it is correct and efficient.

---
In reply to: [597888108](https://github.com/dotnet/roslyn/pull/51900#discussion_r597888108) [](ancestors = 597888108,597866787,597069643)